### ISSUE TITLE
Not supported diffn_nonstrict

### ISSUE THREAD

**Issue by PatrickTrentin88** [google/or-tools#1737]
Starting from the [unison.zip](https://github.com/google/or-tools/files/3881256/unison.zip) MiniZinc model, I generated the FlatZinc formula using the global constraints directory for **SAT** shipped with the latest version of or-tools:

```
~$ mzn2fzn -O- -G mznlib_sat unison.mzn mips_gobmk.helpers.dragon_weak.dzn --output-to-stdout > unison.fzn
```

The generated file is also attached to this report.

This is the output I get from `fzn-or-tools`:

```
 Not supported diffn_nonstrict
*** Check failure stack trace: ***
    @     0x7f7df2a1825a  google::LogMessage::Fail()
    @     0x7f7df2a18195  google::LogMessage::SendToLog()
    @     0x7f7df2a17ac6  google::LogMessage::Flush()
    @     0x7f7df2a1b546  google::LogMessageFatal::~LogMessageFatal()
    @     0x7f7df2b5b1e3  operations_research::sat::(anonymous namespace)::CpModelProtoWithMapping::FillConstraint()
    @     0x7f7df2b5df84  operations_research::sat::SolveFzWithCpModelProto()
    @     0x55991cd7b974  main
    @     0x7f7df26031e3  __libc_start_main
    @     0x55991cd7be0e  _start
    @              (nil)  (unknown)
Aborted (core dumped)
```

The global constraints file `diffn_nonstrict.mzn` is overridden by `ortools` to specifically avoid to translate this constraint with simpler constraints. So why is it not supported? How can I fix this and other, similar, errors?

**Comment by lperron** [comment]
Are you using 7.4 ? It was fixed after the challenge results.
We did actually crash on 20% of the instances :-(

**Comment by PatrickTrentin88** [comment]
I am using the latest release available, `7.4.7247`. If you can provide the binary for the `x86_64` platform without the bug, I may use it in my experiment. The data is going to be included in a paper.

**Comment by lperron** [comment]
You can build from source.
Laurent Perron | Operations Research | lperron@google.com | (33) 1 42 68 53
00



Le lun. 25 nov. 2019 à 14:43, PatrickTrentin88 <notifications@github.com> a
écrit :
